hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -481,7 +479,7 @@ private SnapshotDiffReportOzone createPageResponse(
   private void checkReportsIntegrity(final SnapshotDiffJob diffJob,
                                      final int totalDiffEntries)
       throws IOException {
-    if (diffJob.getTotalDiffEntries() != totalDiffEntries) {
+    if (diffJob.getTotalDiffEntries() > totalDiffEntries) {
```

Comment:
`!=` was correct because it should be neither lower nor higher. Has to exactly equal.

Reply:
Done. Here there was a bug which could mark the snapshot as failed if the client passes the wrong token. I believe we don't validate the token's validity anywhere. Made the changes to fix it. @hemantk-12 @smengcl please take a second look at this if I am wrong.

Reply:
There is [token validation](https://github.com/apache/ozone/blob/9034434285f3877246b3ebc6a7513c082931b279/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java#L706) and [pageSize validation](https://github.com/apache/ozone/blob/9034434285f3877246b3ebc6a7513c082931b279/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java#L667).

Request validation is done in `OmSnapshotManager` on server side.

Reply:
Ah, I see what you meant.

For that, we need to check that `index` passed by client is less than` totalDiffEntries` in `createPageResponse` not here.  This is very basic data integrity to make sure that complete report was returned to client assuming that client fetches whole report and not from in-between.

Check 439-444 is redundant since that already happens in `OmSnapshotManager`. Change it to `if (index >= snapDiffJob.getTotalDiffEntries)` and return empty response with null next token instead of throwing an exception.

Reply:
The check is a good to have if some other function calls this function. We should check this thing it could so happen that the particular entry could exist for some reason in the db. In this case we should ideally mark the report as corrupted.